UnicornRender/video/source/Mesh.cpp

```diff
@@ -10,9 +10,20 @@ namespace unicorn
 {
 namespace video
 {
-Mesh::Mesh(Material material)
-    : m_material(material)
+Mesh::Mesh() :
+    m_material(new Material)
+    /* Here we initialize with new Material
+       When we will call SetMaterial first time in constructor,
+       m_material will be nullptr and m_material->DataUpdated.disconnect will crash. */
```

Comment:
There should be a check in `SetMaterial` because user can pass `nullptr` and it would then break the next `SetMaterial`. Explicit checks are OK. If you're worried about performance impact you can use `assert` instead so in release they will be omitted